pkg/compact/downsample/downsample.go

```diff
@@ -153,9 +155,16 @@ func Downsample(
 			for _, c := range chks {
 				ac, ok := c.Chunk.(*AggrChunk)
 				if !ok {
-					return id, errors.Errorf("expected downsampled chunk (*downsample.AggrChunk) got %T instead for series: %d", c.Chunk, postings.At())
+					if c.Chunk.NumSamples() == 0 {
+						// Downsampled block can erroneously contain empty XOR chunks, skip those
+						// https://github.com/thanos-io/thanos/issues/5272
+						level.Warn(logger).Log("msg", fmt.Sprintf("expected downsampled chunk (*downsample.AggrChunk) got an empty %T instead for series: %d", c.Chunk, postings.At()))
+					} else {
+						return id, errors.Errorf("expected downsampled chunk (*downsample.AggrChunk) got a non-empty %T instead for series: %d", c.Chunk, postings.At())
```

Comment:
Just wondering if we encounter a non-empty XOR chunk then an error is returned. But in this case, what can we do to repair this block currently? If there is no way to fix it can we have a flag to skip this chunk as well?

Reply:
Yea, that's kind of by design - we want to ensure we catch problems instead of ignoring them. But yea, I would not oppose extra flag for it.

Reply:
we could do that in next PR

Reply:
agree, thanks

Reply:
Yeah, let me have another issue to track this. I still want to have a way to fix it somehow in the repair tool because we cannot do anything to fix the error chunk now.